windows/src/desktop/kmshell/install/Keyman.Configuration.UI.KeymanProtocolHandler.pas

```diff
@@ -84,7 +85,7 @@ function TKeymanProtocolHandler.DoHandle(Owner: TComponent;
   // TODO: refactor into separate unit together with code from UfrmInstallKeyboardFromWeb
   FTempDir := IncludeTrailingPathDelimiter(CreateTempPath);  // I1679
   try
-    FDownloadFilename := FTempDir + PackageID + Ext_PackageFile;
+    FDownloadFilename := FTempDir + TIdURI.URLDecode(PackageID) + Ext_PackageFile;
```

Comment:
Ideally `PackageID` is decoded when set and then re-encoded where needed (e.g. line 89). Otherwise we should rename the variable to `PackageIDURIEncoded` or similar to clarify.